pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// nolint:typecheck,goconst,gocritic,stylecheck,nolintlint
+package shared
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kappctrl "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kapppkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+
+	addonutil "github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+const (
+	getResourceTimeout  = time.Minute * 1
+	waitForReadyTimeout = time.Minute * 20
+	pollingInterval     = time.Second * 30
+)
+
+// create cluster client from kubeconfig
+func createClientFromKubeconfig(exportFile string, scheme *runtime.Scheme) (client.Client, error) {
+	config, err := clientcmd.LoadFromFile(exportFile)
+	Expect(err).ToNot(HaveOccurred(), "Failed to load cluster Kubeconfig file from %q", exportFile)
+
+	rawConfig, err := clientcmd.Write(*config)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create raw config ")
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+func checkClusterCBS(ctx context.Context, mccl, wccl client.Client, mcClusterName, wcClusterName, infrastructureName string) error {
+	log.Infof("Verify addons on workload cluster %s", wcClusterName)
+	var err error
+
+	// Get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, mccl, constants.TkgNamespace, mcClusterName)
+
+	// get package details from clusterBootstrap
+	cniPkgShortName, cniPkgName, cniPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)
+	Expect(err).NotTo(HaveOccurred())
+	kappPkgShortName, kappPkgName, kappPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.Kapp.RefName)
+	Expect(err).NotTo(HaveOccurred())
+
+	// verify clusterBootstrap details matches package install on workload cluster
+	verifyPackageInstall(ctx, wccl, wcClusterName, cniPkgShortName, cniPkgName, cniPkgVersion)
+	verifyPackageInstall(ctx, wccl, wcClusterName, kappPkgShortName, kappPkgName, kappPkgVersion)
```

Comment:
Since kapp-controller is a remote packageInstall i.e. the packageInstall for kapp-controller will be on management-cluster. So passing mccl (management cluster client) instead of wccl should make this test pass

Reply:
Oh, good catch! This has been updated